packages/docusaurus-theme-search-algolia/src/theme/SearchBar/index.tsx

```diff
@@ -173,14 +178,19 @@ function DocSearch({
   const transformItems = useRef<DocSearchModalProps['transformItems']>(
     (items) =>
       items.map((item) => {
+        // Replace parts of the URL if the user has configured it in the config
+        const itemUrl = replaceInItemUrl
+          ? item.url.replace(replaceInItemUrl.from, replaceInItemUrl.to)
+          : item.url;
+
```

Comment:
Not super fan of this public api + implementation

What if I want to provide multiple replaces, multiple substitution values and more power with regexes? I somehow have a use case for this myself so I'd rather provide a more flexible API

What about this instead?

```jsx
function replacePathname(pathname, replacements) {
  return replacements.reduce(
    (str, { from, to }) => str.replaceAll(from, to),
    pathname
  );
}

const result = replacePathname("/docs/abc/docs/end/def/end", [
  { from: /^\/docs\//g, to: "/test/" },
  { from: /end$/g, to: "end2" }
]);

console.log(result);
// /test/abc/docs/end/def/end2
```

Regexes can be provided through Node.js config as strings and created with `new RegExp(str, "g")`

Ideally I with we should just provide a function for maximum flexibility but unfortunately it's not serializable

Reply:
Note: shipping a single replacement is probably good enough for now as long as we can do things like `(docs/version1|docs/version2)` with a regexp

We can add array of replacements support later eventually if someone needs this. If the themeConfig name can reflect that the value might later support both single/multi replacements, that's better as we won't have to rename that config option attribute